### PR TITLE
feat(collector): added support for redis v4

### PR DESCRIPTION
**Pull request by kirrg001**
refs 104481

https://github.com/redis/node-redis/releases/tag/redis%404.0.0
https://github.com/redis/node-redis#supported-redis-versions

- [x] redis has a sub/pub feature, which is not tested/covered. I would like to create a new ticket for this (105777)
  - https://github.com/redis/node-redis/tree/v0.12.1#publish--subscribe
  - https://github.com/redis/node-redis/tree/v3.1.2#pubsub
  - https://github.com/redis/node-redis/tree/v4.0.0-rc.3#pubsub

### CODE REVIEW

**Review by basti1302 — APPROVED** [review]
LGTM! I left some minor remarks. 😼

File: packages/collector/test/tracing/database/redis/test.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const ProcessControls = require('../../../test_util/ProcessControls');
 const globalAgent = require('../../../globalAgent');
 
 describe('tracing/redis', function () {
-  ['latest', 0].forEach(redisVersion => {
+  ['latest', 'v3', 'v0'].forEach(redisVersion => {
```
Comment (basti1302) [inline packages/collector/test/tracing/database/redis/test.js:25]: @kirrg001 Just saw that (after submitting my review) - looks like we might need to disable tests for `latest` on Node.js 10. The 4.0 [changelog](https://github.com/redis/node-redis/blob/master/CHANGELOG.md#breaking-changes) says Node.js 12 is the minimum version now.

**Review by basti1302 — APPROVED** [review]

File: packages/collector/test/tracing/database/redis/test.js (hunk 1 above)
Comment (kirrg001) [reply packages/collector/test/tracing/database/redis/test.js:25]: Good spot 🙏

File: packages/core/src/tracing/instrumentation/database/redis.js [hunk 2]
```diff
@@ -32,103 +27,237 @@ exports.deactivate = function deactivate() {
 };
 
 exports.init = function init() {
+  // v4 commands, "redis-commands" is outdated and no longer compatible with it
```
[the remark this answers is not in the document]
Comment (kirrg001) [reply packages/core/src/tracing/instrumentation/database/redis.js:30]: Will change 👍

**Pull request merged**